### PR TITLE
fetch: added in test cases for various constraints

### PR DESCRIPTION
**Pull request by ryanluu12345**
Increased test coverage to include other cases that can result from improper target schema creation, especially around constraints. Makes sure that these get recorded in the exceptions logging table.

Referenced this: https://github.com/cockroachdb/cockroach/blob/786ed22e6c319b8d28399f8ee1f82b88efe7c1ef/pkg/sql/pgwire/pgcode/codes.go#L326

Release Note: None

### CODE REVIEW

File: fetch/testdata/pg/exceptions.ddt
Comment (Jeremyyang920) [inline fetch/testdata/pg/exceptions.ddt:1]: Can we split the different tests up a bit more so this exceptions.ddt file doesnt become so long and unreadable after a certain point?

Comment (ryanluu12345) [reply fetch/testdata/pg/exceptions.ddt:1]: Sure, let me split the new tests up into constraint-exceptions.ddt

Comment (ryanluu12345) [reply fetch/testdata/pg/exceptions.ddt:1]: @Jeremyyang920 how are the latest changes?

**Review by Jeremyyang920 — APPROVED** [review]

File: fetch/testdata/pg/exceptions.ddt
Comment (ryanluu12345) [reply fetch/testdata/pg/exceptions.ddt:1]: Yeah I agree. We can always split them out to their own test files in the future. I don't have a problem with that. Let's see how this feature evolves with continuation work first.

**Pull request merged**